src/Pagination.tsx

```diff
@@ -588,9 +588,11 @@ const Pagination: React.FC<PaginationProps> = (props) => {
         selectPrefixCls={selectPrefixCls}
         changeSize={showSizeChanger ? changePageSize : null}
         pageSize={pageSize}
-        pageSizeOptions={pageSizeOptions}
+        pageSizeOptions={(typeof showSizeChanger === 'object' && showSizeChanger?.options) || pageSizeOptions}
         quickGo={shouldDisplayQuickJumper ? handleChange : null}
         goButton={gotoButton}
+        onChange={typeof showSizeChanger === 'object' && showSizeChanger?.onChange}
```

Comment:
changeSize and onChange is basicly same function, we should unify them to one

Reply:
@afc163 Can I remove onChange from showSizeChanger?

Reply:
LGTM, Pagination's `onChange` could be enough in most cases.